### PR TITLE
Workaround ENTESB-17802 by managing camel-kamelet-reify explictitly …

### PR DESCRIPTION
**Pull request by ppalaga**
by managing camel-kamelet-reify explictitly before importing camel-bom and camel-quarkus-bom

https://issues.redhat.com/browse/ENTESB-17802

### CODE REVIEW

**Comment by heiko-braun** [comment]
@ppalaga What the impact of this change? Do you expect us the rebuild camel k for this? 
//cc @balejosg @squakez

**Comment by squakez** [comment]
> @ppalaga What the impact of this change? Do you expect us the rebuild camel k for this? //cc @balejosg @squakez

I think a rebuild of camel-k-runtime (so, definetely camel-k) is needed whichever is the change we apply on dependencies.

**Comment by ppalaga** [comment]
I have added some background info to https://issues.redhat.com/browse/ENTESB-17802?focusedCommentId=19302095&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-19302095

**Review by squakez — APPROVED** [review]

**Comment by ppalaga** [comment]
Looking at https://github.com/apache/camel-k/blob/main/pkg/builder/quarkus.go#L115-L131 an additional fix is needed in camel-k-runtime-bom . Let me prepare it.
Except for that the order of bom imports in https://github.com/apache/camel-k/blob/main/pkg/builder/quarkus.go#L115-L131 needs to be switched so that camel-k-runtime-bom is first. After that, all user apps will get the prod version of camel-kamelet-reify

**Comment by ppalaga** [comment]
> Looking at https://github.com/apache/camel-k/blob/main/pkg/builder/quarkus.go#L115-L131 an additional fix is needed in camel-k-runtime-bom .

Done in 8abae53 . Please re-review.

File: support/camel-k-runtime-bom/pom.xml [hunk 1]
```diff
@@ -32,8 +32,9 @@
     <packaging>pom</packaging>
 
     <properties>
-        <camel-version>3.11.0</camel-version>
-        
+        <camel-version>3.11.1.fuse-800003-redhat-00002</camel-version>
```
Comment (squakez) [inline support/camel-k-runtime-bom/pom.xml:35]: That one should be a placeholder eventually replaced by the productization process. @cunningt, @svkcemk can you confirm if during the build also this file will checked and the version replaced accordingly?

**Comment by ppalaga** [comment]
> @ppalaga What the impact of this change? Do you expect us the rebuild camel k for this?

Yes, this requires a rebuild of Camel K Runtime and Camel K. I heart that a rebuild is planned for other reasons so I hope this fix is not a big issue.

File: support/camel-k-runtime-bom/pom.xml (hunk 1 above)
Comment (cunningt) [reply support/camel-k-runtime-bom/pom.xml:35]: The dev build will replace this with 3.11.1.fuse-800025 and then the PNC build will replace it with 3.11.1.fuse-800025-redhat-00001.   I'm not sure this works.

Comment (ppalaga) [reply support/camel-k-runtime-bom/pom.xml:35]: > The dev build will replace this with 3.11.1.fuse-800025 and then the PNC build will replace it with 3.11.1.fuse-800025-redhat-00001.

Yes, that's fine, exactly as expected.

Comment (squakez) [reply support/camel-k-runtime-bom/pom.xml:35]: According this script (and the comment), it seems to be done only on the parent pom: https://github.com/jboss-fuse/fuse-jenkins-pipelines/blob/main/8.5/pipelines/camelk/Jenkinsfile#L147-L148 is that correct?

**Comment by squakez** [comment]
Merged. We may still need to adjust the productization process to replace the placeholder in that bom file @cunningt

**Pull request merged**